libbeat/outputs/elasticsearch/api_integration_test.go

```diff
@@ -82,6 +86,9 @@ func TestIngest(t *testing.T) {
 	}
 
 	_, _, err := client.DeletePipeline(pipeline, nil)
+	if err != nil && err.Error() != statusNotFound {
```

Comment:
Comparing error stings is brittle, I would use it only when there are no other options. In this case we already have the http status code as an `int` so I think you could use that in combination with the fact that an error occurred.